tests/explainers/test_kernel.py

```diff
@@ -360,3 +360,16 @@ def test_kernel_logits_zeros_ones_probs(nsamples):
     pred = rf.predict_proba(X_test_sampled)
 
     np.testing.assert_allclose(sigm(shap_values.values.sum(1) + explainer.expected_value), pred, atol=1e-04)
+
+
+@pytest.mark.parametrize("dt", [np.bool_, np.object_])
```

Comment:
This test is a bit weird in the sense that it doesn't really "fail before" and "pass after," but it does fail if the fallback I recently added uses `return 0 if i == j else 1` instead of `return 0 if all(i == j) else 1` because for the `np.object_` `dtype` you'll get the typical `ValueError: The truth value of an array with more than one element is ambiguous`.

So it is more of a sniff test for things slipping through the cracks, but it isn't super picky beyond that, and in fact the original returning of `None` noted above in some cases still allows the test to pass. It may be possible to cook the test up in a way that is more stringent than that, but it is at least doing *something*.